chore(docs): refine the "getting started with modeling" page

pavokta opened the pull request:
This is a first-half edit of a very, very long document.

I know that this is a lot to look over, so feel free to schedule me for a Zoom call so we can talk edits over before you merge.

<!-- Thanks for opening a PR! Here are some quick tips:
If this is your first time contributing, [read our Contributing Guidelines](https://github.com/openfga/.github/blob/main/CONTRIBUTING.md) to learn how to create an acceptable PR for this repo.
By submitting a PR to this repository, you agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md)

If your PR is under active development, please submit it as a "draft". Once it's ready, open it up for review.
-->

<!-- Provide a brief summary of the changes -->

## Description
<!-- Provide a detailed description of the changes -->

## References
<!-- Provide a list of any applicable references here (Github Issue, [OpenFGA RFC](https://github.com/openfga/rfcs), other PRs, etc..) -->

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

pavokta committed:
Update getting-started.mdx

This is a first-half edit of a very, very long document.

I know that this is a lot to look over, so feel free to schedule me for a Zoom call so we can talk edits over before you merge.

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -83,17 +72,13 @@ When you are modeling, you need to answer a more general question:
-If you can answer that question for all types of objects in your system, then you can codify that into an authorization model.
-
-Let's get started!
+Answering that question for all object types in your system codifies your authorization model.
```
I don't think this is saying the same thing as the previous sentence.

It used to say, finish these steps then you can write your authorization model, now it says: finishes these steps and you would have written your authorization model

pavokta replied:
How about:

Answering that question for all object types in your system means your authorization model can be codified.

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -83,17 +72,13 @@ When you are modeling, you need to answer a more general question:
-Defining an authorization model requires codifying an answer to the question "why could user U perform an action A on an object O?" for all use cases or actions in your system. This is an iterative process. For the purpose of this guide, we'll go through one iteration of this process using a simplified Google Drive like system as an example.
-
-Steps for defining your authorization model:
+Below is one iteration of the codification process, using a simplified Google Drive-like system as an example. The following steps define your authorization model:
```
maybe codification is the wrong term here, can we replace it with something else

```suggestion
Below is one iteration of defining an authorization model for your system, using a simplified Google Drive-like system as an example. The following are the steps for this iteration:
```

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -354,28 +336,24 @@ Now that we have a list of object types we can start defining them using the <Up
-
 :::info Important
-In a few cases other users can be part of determining whether an action can be performed on an action or not. Social media is an example of this "a user can comment on a picture if they are a friend of the user that published it".
+Though you're in the process of building a version you can use, the model above is not yet a valid authorization model accepted by <ProductName format={ProductNameFormat.ShortForm}/>.
```
```suggestion
```

This is no longer true

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -354,28 +336,24 @@ Now that we have a list of object types we can start defining them using the <Up
-In a few cases other users can be part of determining whether an action can be performed on an action or not. Social media is an example of this "a user can comment on a picture if they are a friend of the user that published it".
+Though you're in the process of building a version you can use, the model above is not yet a valid authorization model accepted by <ProductName format={ProductNameFormat.ShortForm}/>.
+
+In a few cases, other users can determine whether one action can be performed on another action. For example, on social media, sometimes a user can comment on a picture if they are a friend of the user that published it.
```
I don't think this whole note is helpful here, let's remove it. It used to be back in back in the day where we didn't require a type for an object

pavokta replied:
Fair enough!

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -354,28 +336,24 @@ Now that we have a list of object types we can start defining them using the <Up
+In a few cases, other users can determine whether one action can be performed on another action. For example, on social media, sometimes a user can comment on a picture if they are a friend of the user that published it.
 
-In those cases [**User** should also be an object type](./building-blocks/object-to-object-relationships.mdx). Following the last recommendation, we would discover the User type because it is a second noun in an expression: "friend of the user".
+In those cases [**User** must be an object type](./building-blocks/object-to-object-relationships.mdx), and you would discover the User type because it is a second noun in an expression `friend of the user`.
```
Let's remove

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -354,28 +336,24 @@ Now that we have a list of object types we can start defining them using the <Up
 
-- any noun that is the \{noun\} of a "\{noun\} of a/an/the \{type\}" expression. **These are typically the Foreign Keys in a database.** We'll highlight these in <span className="green-highlight-text">green</span>.
-- any verb or action that is the \{action\} of a "can \{action\} (in) a/an \{type\}" expression. **These are typically the permissions for a type.** We'll highlight these in <span className="yellow-highlight-text">yellow</span>.
+- any noun that is the \{noun\} of a "\{noun\} of a/an/the \{type\}" expression. These are typically the Foreign Keys in a database, highlighted below in <span className="green-highlight-text">green</span>.
```
I feel by removing the bold, we're no longer highlighting an important point we want to stress

pavokta replied:
That's not how the Docs style guide recommends **bold** be used. One thing I'm going to do pretty soon is work on a modified style guide for FGA, but for now, I'd like to hew pretty closely to the original style guide.

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -476,10 +454,10 @@ The resulting list is:
 
 :::info
-In <ProductName format={ProductNameFormat.ShortForm}/>, relations can only have alphanumeric characters, underscores and hyphens. We recommend using underscore (\_) to separate words and removing prepositions. E.g.: "can create a document" can become "can_create_document" or "create_document" if you are into brevity.
+In <ProductName format={ProductNameFormat.ShortForm}/>, relations may only use alphanumeric characters, underscores and hyphens. We recommend using underscore (\_) to separate words and removing prepositions, so `can create a document` becomes `can_create_document` or `create_document`.
```
```suggestion
In <ProductName format={ProductNameFormat.ShortForm}/>, relations may only use alphanumeric characters, underscores and hyphens. We recommend using underscore (\_) to separate words and removing prepositions, so `can create a document` becomes `can_create_document`.
```

Let's just stick to the first

rhamzeh commented on docs/content/modeling/getting-started.mdx:
```diff
@@ -476,10 +454,10 @@ The resulting list is:
 :::
 
-Using the <UpdateProductNameInLinks link="../configuration-language" name="{ProductName} Configuration Language" /> we can enumerate the relations for each type:
+Enumerate relations for each type with the<UpdateProductNameInLinks link="../configuration-language" name="{ProductName} Configuration Language" />:
```
```suggestion
Enumerate relations for each type with the <UpdateProductNameInLinks link="../configuration-language" name="{ProductName} Configuration Language" />:
```

pavokta committed:
Update docs/content/modeling/getting-started.mdx

Co-authored-by: Raghd Hamzeh <raghd.hamzeh@openfga.dev>

pavokta committed:
Update docs/content/modeling/getting-started.mdx

Co-authored-by: Raghd Hamzeh <raghd.hamzeh@openfga.dev>

pavokta committed:
Update docs/content/modeling/getting-started.mdx

Co-authored-by: Raghd Hamzeh <raghd.hamzeh@openfga.dev>

pavokta committed:
Update docs/content/modeling/getting-started.mdx

Co-authored-by: Raghd Hamzeh <raghd.hamzeh@openfga.dev>

pavokta committed:
Update docs/content/modeling/getting-started.mdx

Co-authored-by: Raghd Hamzeh <raghd.hamzeh@openfga.dev>

ryanpq committed:
Updated language recommended by @pavokta

Change as specified at https://github.com/openfga/openfga.dev/pull/656#discussion_r1523717725
